server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -291,7 +294,8 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(finalReduce);
         }
         out.writeBoolean(ccsMinimizeRoundtrips);
-        if (out.getTransportVersion().before(TransportVersions.REMOVE_MIN_COMPATIBLE_SHARD_NODE)) {
+        if (out.getTransportVersion().before(TransportVersions.REMOVE_MIN_COMPATIBLE_SHARD_NODE)
+            || out.getTransportVersion().onOrAfter(TransportVersions.REVERT_REMOVE_MIN_COMPATIBLE_SHARD_NODE)) {
```

Comment:
I think we are missing a writeBoolean here? What used to be `out.writeBoolean(minCompatibleShardNode != null);`
Maybe we want to always write false.. IDK. Seems strange to me we are not writing anything though (and we read it)

Reply:
The `writeBoolean` is the line below

Reply:
Oh ok got it. I was confused by the `before`